### PR TITLE
feat: get product & productOptions by brand, brandLocations & location

### PR DESCRIPTION
**Pull request by jayant13o6**
## Description
issues resolved: #975 and #948 
1)When a product/ product option is published it should be visible on website otherwise not
2)When a product / product option is unavailable its opacity will be decreased and a button "Out Of Stock" added in bottom to easily identify it.
3)If a user still clicks on an unavailable product a new popup would open which mentions the product name and image with the "Out Of Stock" tag.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] Database schema is updated
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
Affected Apps
- [ ] Store
    - [ ] SEO
    - [ ] Menu
    - [ ] Checkout
    - [ ] Add to Cart logic
    - [ ] Location Logic
    - [x] Product page
    - [ ] Login
    - [ ] Payment
    - [ ] Kiosk

### CODE REVIEW

**Comment by jayant13o6** [comment]
https://docs.google.com/document/d/1f7quFz2xK_P8K9VG6JlbtO7YJnvifTXLd7Och3X5efI/edit?usp=sharing

File: store/src/components/modifierPopUpForUnAvailability.jsx [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+                         <CloseIcon size={20} stroke="currentColor" />
+                      </div> */}
+                   </div>
+                </div>
+             )}
+          </div>
+       </>
+    )
+ }
+
```
Comment (rahul72925) [inline store/src/components/modifierPopUpForUnAvailability.jsx:306]: remove unnecessary content (as we show only product name, product price and out of stock button there should be only three things)

File: store/src/components/modifier_popup.jsx [hunk 2]
```diff
@@ -655,6 +657,8 @@ export const ModifierPopup = props => {
                         {productOptionsGroupedByProductOptionType
                            .find(eachType => eachType.type == productOptionType)
                            .data.map(eachOption => {
+                              if(!eachOption.isPublished){return null}
+                              else{
```
Comment (rahul72925) [inline store/src/components/modifier_popup.jsx:661]: need not to add `else`. bcz this is only condition left to execute.

File: store/src/components/modifierPopUpForUnAvailability.jsx [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+                   style={{ padding: '0 32px' }}
+                   className="hern-modifier-popup-add-to-cart-btn-parent-div"
+                >
+                   {showCounterBtn && counterButtonPosition == 'BOTTOM' && (
+                   
+                   <Button
+                      className="hern-product-modifier-pop-up-add-to-cart-btn"
+                      style={{ padding: '16px 0px 34px 0px' }}
+                      disabled={
+                         locationId ? (storeStatus.status ? false : true) : true
```
Comment (rahul72925) [inline store/src/components/modifierPopUpForUnAvailability.jsx:275]: this button will always disable bcz we are showing this file on product out of stock.

File: store/src/components/modifier_popup.jsx [hunk 4]
```diff
@@ -671,16 +675,20 @@ export const ModifierPopup = props => {
                                        display: 'flex',
                                        justifyContent: 'space-between',
                                        marginBottom: '8px',
-                                       cursor: 'pointer',
+                                       cursor: `${!eachOption.isAvailable ? '' : 'pointer'}`,
```
Comment (rahul72925) [inline store/src/components/modifier_popup.jsx:678]: make cursor `not-allowed` instead of `''`

File: store/src/components/product_card.jsx [hunk 5]
```diff
@@ -341,7 +341,30 @@ export const ProductCard = props => {
             </div>
          )}
          {showModifier && data && (
-            <ModifierPopup
+            <>{
+            (!data.isAvailable ||
+               !data.productOptions.some(
+                  ele => ele.isAvailable === true && ele.isPublished === true
+               )) ?
+            <ModifierPopupForUnAvailability
```
Comment (rahul72925) [inline store/src/components/product_card.jsx:354]: remove `ModifierPoupupForUnAvailability` component from `product_cart` component and add this component directly on product page file where we are rendering `modifier_popup` component.
make that rendering conditionally and render `ModifierPoupupForUnAvailability`.
[here](https://github.com/dailykit/HERN/blob/3b066d97f0f9cf585b212a1b6bbb635ea2342667/store/src/sections/product/index.js#L148)

**Comment by shubhamz950** [comment]
@jayant13o6 is everything fixed on this branch ?

**Pull request merged**